pkg/plugins/validation.go

```diff
@@ -0,0 +1,121 @@
+package plugins
+
+import (
+	"embed"
+	"path"
+
+	"github.com/ghodss/yaml"
+	"github.com/xeipuuv/gojsonschema"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/kumahq/kuma/pkg/core/validators"
+)
+
+//go:embed policies/*/api/v1alpha1/schema.yaml
+var content embed.FS
+
+var resourceToSchema map[string]*gojsonschema.JSONLoader
+
+func ValidateResourceSchema(message proto.Message, type_ string) error {
```

Comment:
If we call this only from `rest.Unmarshal` then I'd pass marshaled JSON as a first arg, to avoid extra marshaling in this function:
```go
func ValidateResourceSchema(resource []byte, type_ string) error {
  return validateSchema(string(json), type_)
}
```
and after that `validateSchema` method could be inlined

Reply:
nit: to avoid `[]byte` to `string` conversion you can use `gojsonschema.NewBytesLoader(document)` in `validateSchema`

Reply:
👍

Reply:
done